graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/aggregations/AggregationsFetcherSupport.java

```diff
@@ -0,0 +1,222 @@
+import static io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations.SupportedAggregationFunction.MIN;
+import static io.stargate.graphql.schema.cqlfirst.dml.fetchers.aggregations.SupportedAggregationFunction.SUM;
+
+import graphql.schema.DataFetchingEnvironment;
+import graphql.schema.SelectedField;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.query.builder.QueryBuilder;
+import io.stargate.db.schema.Table;
+import io.stargate.graphql.schema.cqlfirst.dml.NameMapping;
+import io.stargate.graphql.schema.cqlfirst.dml.fetchers.DbColumnGetter;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+public class AggregationsFetcherSupport {
+
+  private final Table table;
+  private final DbColumnGetter dbColumnGetter;
+
+  public AggregationsFetcherSupport(NameMapping nameMapping, Table table) {
+    this.table = table;
+    this.dbColumnGetter = new DbColumnGetter(nameMapping);
+  }
+
+  public void addAggregationFunctions(
+      DataFetchingEnvironment environment, QueryBuilder.QueryBuilder__20 queryBuilder) {
+    List<SelectedField> valuesFields = environment.getSelectionSet().getFields("values");
+    if (valuesFields.isEmpty()) {
+      return;
+    }
+
+    for (SelectedField valuesField : valuesFields) {
+      for (SelectedField selectedField : getAllFieldsWithNameArg(valuesField)) {
+        Map<String, Object> arguments = selectedField.getArguments();
+        SupportedAggregationFunction functionName = getNameArgument(arguments);
+        switch (functionName) {
+          case COUNT:
+            addAggregation(
+                arguments,
+                queryBuilder,
+                QueryBuilder.QueryBuilder__20::count,
+                selectedField,
+                COUNT);
+            break;
+          case AVG:
+            addAggregation(
+                arguments, queryBuilder, QueryBuilder.QueryBuilder__20::avg, selectedField, AVG);
+            break;
+          case MIN:
+            addAggregation(
+                arguments, queryBuilder, QueryBuilder.QueryBuilder__20::min, selectedField, MIN);
+            break;
+          case MAX:
+            addAggregation(
+                arguments, queryBuilder, QueryBuilder.QueryBuilder__20::max, selectedField, MAX);
+            break;
+          case SUM:
+            addAggregation(
+                arguments, queryBuilder, QueryBuilder.QueryBuilder__20::sum, selectedField, SUM);
+            break;
+        }
+      }
+    }
+  }
+
+  public Map<String, Object> addAggregationResults(
+      Map<String, Object> columns, DataFetchingEnvironment environment, Row row) {
+    List<SelectedField> valuesFields = environment.getSelectionSet().getFields("values");
+    if (valuesFields.isEmpty()) {
+      return columns;
+    }
+    for (SelectedField valuesField : valuesFields) {
+      for (SelectedField selectedField : getAllFieldsWithNameArg(valuesField)) {
```

Comment:
It is needed because the `SupportedGraphqlFunction.valueOfIgnoreCase`
throws if the name of the function is not found:
```
 throw new UnsupportedOperationException(
        String.format("The graphQL aggregation function: %s is not supported.", functionName));
```
I can change it to a less strict approach, and return optional from it. Next, execute the `putResultValue` only if the optional value is present. wdyt?

Reply:
Yeah I think that would be cleaner, there could be other fields with a `name` attribute in the future.

Reply:
Ok, I will add the optional. However, we still need `getAllFieldsWithNameArg` because it provides the deduplication of fields (please note that it returns Set). The `valuesFields` contains a duplicate of every entry.